compiler/qsc_eval/src/backend.rs

```diff
@@ -419,6 +420,19 @@ impl Backend for SparseSim {
                 self.apply_noise(q);
                 Some(Ok(Value::unit()))
             }
+            "Apply" => {
+                let [matrix, qubits] = unwrap_tuple(arg);
+                let qubits = qubits
+                    .unwrap_array()
+                    .iter()
+                    .filter_map(|q| q.clone().unwrap_qubit().try_deref().map(|q| q.0))
+                    .collect::<Vec<_>>();
+                let matrix = unwrap_matrix_as_array2(matrix, &qubits);
```

Comment:
I need to add another check here to make sure the matrix is unitary in it's numerical contents. If someone passes in a matrix that causes the state vector to be empty, like the all zero matrix, then it will panic the simulator.

Reply:
Added a check here that uses the `ndarray` functionality to verify the constraints.